### DISCUSSION TITLE
App idea for entire library export

### DISCUSSION THREAD

**Discussion by cfc62** [RhetTbull/osxphotos#1364] [Ideas]
Given a decent amount of commentary that I see repeatedly, a common concern that a lot of people using Photos share is the fear of being locked into Photos. 

At some level, I share this concern as well. 

Many believe their photos go into some massive database scheme and once they go in, they can't come back out. We know this isn't true, but it's a very common misconception. It doesn't help that Apple is a bit weird about exporting. 

To wit, if there were an app that just did one thing - export all your originals, either with or without edits - I believe that app would have takers - and it would probably get people more comfortable into committing to Photos. 

To be clear, when I say "app" I mean a GUI-based app. While I'm more than comfortable with scripting and programming, I know I'm in the minority. But even so, I'd love an easy-to-use hard-to-screw-up library export app. 

The usual niceties would apply, making sure you have enough space on the target device, smartly building subdirectories if needed, restoring original filenames, maintaining metadata, exporting edited versions if requested, giving the user a progress bar with ETA and so on. 

I know I'd pay for an app like this - from a trusted source, of course.

**Comment by RhetTbull** [comment]
I have considered building such an app but would need to learn some additional GUI skills first and I just don't have time at the moment. I am working on a simple menu bar GUI app that will run a saved osxphotos config regularly to maintain a backup without using the command line. This will be fairly simple but I plan to try to monetize it as a way to support osxphotos development. Someday I'd like to create a fully GUI version of osxphotos.

There is a [Photos Takeout](https://photostakeout.com/) app that purports to do basically what you outlined. I'm not really familiar with it but have used it just to see what the "competition" was like -- it's very basic but might work for the average user. I do very much want to create my own version using what I've learned from osxphotos but that will take some time.

**Reply by RhetTbull** [reply]
@cfc62 I've been thinking about this and I think I can implement such an app as part of the GUI I'm bulding for scheduling osxphotos runs. So it would be a status bar app that can run a single one-time export or allow that export to be scheduled for periodic execution. 

What osxphotos features do you think are important for such an app? The more configuration that's added , the more complex the app and the interface.

How much do you think an app like this would be worth?

**Reply by cfc62** [reply]
Overall, simplicity has a quality all of its own for something like this; there's a real risk of putting too many things into the app, as you note. 

Having said that, there is _some_ flexibility and safety net that's needed:

Flexibility:
- Ability to choose any valid photolibrary to export
- Ability to choose any storage device that has enough space
- Ability to choose if only true originals are exported, or if assets resulting from edits are exported, or both
- Ability to choose to retain metadata (I know some metadata embeds itself into the original, but not all)
- RAW and JPG plus HEIF and other formats handled appropriately
- Ability to choose from some sort of folder naming convention* for the export

Safety net:
- Ensuring that the target device/area has sufficient space for the export to complete
- % done/elapsed time/ETA for completion (hard to do the ETA, but helpful)
- Avoiding copy-on-write issues if exporting to the same APFS volume as the photolibrary being exported from
- Helpful user assistance to make the right decisions for the choices in the "flexibility" choices above

* So for folder naming/structure convention I would tend to think that year-month-day-hour by capture time would fit the bill, but that's just like my opinion, man. 

**2022** folder contains... a folder for each month where there are applicable assets
**September** folder contains... a folder for each date...
**17** folder contains... a folder for each hour
**10pm** folder contains... the originals for that time. 

Even with this, one might have to subdivide hourly folders such that 1000 images don't all wind up in the same directory; even to this day most OSs will not like that many files in one directory.

Then again, maybe an organization based on import session might make more sense for other people. Not pretending I have the answers here. 

A single-purpose app like this would be in the $5-15-20 range, depending on features/functionality. Perhaps it starts at the low end and migrates higher over time (people hate this though, but a real killer is charging more than the value right out of the gate). 

I'd pay more if my edits were exported. While I don't have a huge amount, it's not something I want to walk away from either.
